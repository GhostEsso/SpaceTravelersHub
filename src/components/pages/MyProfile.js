import { useSelector } from 'react-redux';
import { selectMission } from '../../redux/Mission/MissionSice';

const MyProfile = () => {
  // Make sure you're using the correct slice name here
  const { rockets } = useSelector((state) => state.rockets);
  const { missionData, joinedMissions } = useSelector(selectMission);
  const missionID = joinedMissions;

  const Missioners = missionData.filter((mission) => missionID.includes(mission.mission_id));

  return (
    <>
      <section className="myProfileContainer">
        <div className="sectionA">
          <h2>My Missions</h2>
          <ul className="rockets">
            {Missioners.map((mission) => (
              <li key={mission.mission_id} className="missioName">
                {mission.mission_name}
              </li>
            ))}
          </ul>
        </div>
        <div className="sectionB">
          <h2>My Rockets</h2>
          <ul className="rockets">
            {rockets
              .filter((rocket) => rocket.reserved)
              .map((rocket) => (
                <li key={rocket.id}>{rocket.rocket_name}</li>
              ))}
          </ul>
        </div>
      </section>
    </>
  );
};

export default MyProfile;
