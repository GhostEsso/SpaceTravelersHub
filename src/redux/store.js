import { configureStore } from '@reduxjs/toolkit';
import Rocketslice from './rockets/RocketsSlice';
import MissionSice from './Mission/MissionSice';

const store = configureStore({
  reducer: {
    rockets: Rocketslice,
    mission: MissionSice,
  },
});

export default store;
