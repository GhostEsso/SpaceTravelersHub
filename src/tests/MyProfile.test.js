import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import MyProfile from '../components/pages/MyProfile';
import store from '../redux/store';
import { it, expect } from '@jest/globals';

it('renders correctly', () => {
  const missions = [
    { id: 1, mission_name: 'Mission 1', reserved: true },
    { id: 2, mission_name: 'Mission 2', reserved: false },
    // Add more mission objects as needed
  ];
  const rockets = [
    { id: 1, rocket_name: 'Mission 1', reserved: true },
    { id: 2, rocket_name: 'Mission 2',  reserved: false },
    // Add more mission objects as needed
  ];

  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <MyProfile missions={missions} rockets={rockets} />
      </BrowserRouter>
    </Provider>
  );

  expect(container).toMatchSnapshot();
});
