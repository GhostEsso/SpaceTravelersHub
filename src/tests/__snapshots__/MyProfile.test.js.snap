// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section
    class="myProfileContainer"
  >
    <div
      class="sectionA"
    >
      <h2>
        My Missions
      </h2>
      <ul
        class="rockets"
      />
    </div>
    <div
      class="sectionB"
    >
      <h2>
        My Rockets
      </h2>
      <ul
        class="rockets"
      />
    </div>
  </section>
</div>
`;
